### PR TITLE
Add endpoint to remove web push subscription

### PR DESCRIPTION
**Pull request by oneiros**
... and include in an HTTP header to be used by the receiving party in case of delivery errors.

Fixes MAS-328

### CODE REVIEW

File: app/controllers/api/web/push_subscriptions_controller.rb [hunk 1]
```diff
@@ -17,6 +17,13 @@ def update
     render json: @push_subscription, serializer: REST::WebPushSubscriptionSerializer
   end
 
+  def destroy
+    push_subscription = ::Web::PushSubscription.find_by_token_for(:unsubscribe, params[:id])
```
Comment (renchap) [inline app/controllers/api/web/push_subscriptions_controller.rb:21]: What happens if we cant find this subscription? I think it should be a no-op?

Comment (oneiros) [reply app/controllers/api/web/push_subscriptions_controller.rb:21]: Good point. This line will actually not fail, but the next would raise an error. I fixed that just now.

File: spec/requests/api/web/push_subscriptions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      Fabricate(:web_push_subscription)
+    end
+    let(:token) do
+      web_push_subscription.generate_token_for(:unsubscribe)
+    end
+
+    it 'deletes the subscription' do
+      expect { subject }
+        .to change(Web::PushSubscription, :count).by(-1)
+    end
```
Comment (ThisIsMissEm) [inline spec/requests/api/web/push_subscriptions_spec.rb:19]: Probably needs a test case for the "subscription not found" case that renaud pointed out, just to assert it definitely always gives a 200 head response

**Review by ThisIsMissEm — COMMENTED** [review]
This looks good to me, based on what Renaud told me about this, only concern is that extra test case.

File: spec/requests/api/web/push_subscriptions_spec.rb (hunk 2 above)
Comment (oneiros) [reply spec/requests/api/web/push_subscriptions_spec.rb]: Good point. I just added two more test cases.

Comment (ThisIsMissEm) [reply spec/requests/api/web/push_subscriptions_spec.rb]: This seems good to me now, I'll leave it up to y'all as to when/if this gets merged, but implementation wise this looks solid to me.

**Review by Gargron — APPROVED** [review]

**Pull request merged**